### PR TITLE
Complete Building Budget App

### PR DESCRIPTION
**Pull request by NoelLincoln**
Hello 👋,

In this milestone I was able to build my expense tracker as per the requirements:

**Design**
- follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
 Colors, Typography: font face, size and weight.
Layout: composition and space between elements.

**Interactions**
Splash screen:

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (categories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
Testing requirements
Create unit and integration tests for all the most important components of your RoR application.
Technical requirements
You should use Postgres as your database.
You should use devise for authentication.
You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
You can use a view template engine of your choice (.erb, .slim, .haml).
The project should be deployed and accessible online.

### CODE REVIEW

**Comment by Munsa1** [comment]
Hello @NoelLincoln  you have used incorrect git flow, you should request to merge a feature branch into development branch.
Not to worry just rename `main` branch to `development`. But do this on github site not locally. if you need help reach out via slack

**Comment by NoelLincoln** [comment]
Hello Munsa,

I have corrected it.

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                  <div class="col-md-4">
+                    <%= image_tag(category.icon, class: 'img-fluid rounded-start') if category.icon.present? %>
+                  </div>
+                  <p class="card-text link p-text">$<%= category.user_transactions.sum(:amount) %></p>
+                </div>
+              </div>
+
+            </div>
+          </div>
+        <% end %>
```
Comment (thecodechaser) [inline app/views/categories/index.html.erb]: - [ ] The styling in the categories page is broken, All categories are displaying different sizes, and the `add new category` button is overlapping on the categories and it also scrolls up with categories when I scroll the page. Please make sure to fix these issues.

image

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<div class="container mt-5 pt-5">
+  <div class="row">
+    <% @categories.each do |category| %>
+      <div class="col-6"> <!-- Bootstrap class for 2 cards per row on medium screens and up -->
+        <%= link_to category_user_transactions_path(category), class: 'category-link link' do %>
+          <div class="card mb-4">
```
Comment (thecodechaser) [inline app/views/categories/index.html.erb]: - [x] Please make sure that the `add new category` and `add new transaction` buttons styles match with the given design as requested in the project requirements to follow the design guidelines.

### Your design

image

image

### Given design

image

File: spec/features/categories/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.feature 'Categories Index Page - Viewing Transactions', type: :feature do
+  scenario 'User can view category transactions' do
+    user = create_and_sign_in_user
+    category = create_category_for_user(user)
```
Comment (thecodechaser) [inline spec/features/categories/index_spec.rb:6]: - [ ] The below 4 tests are failing, Please make sure that all tests are passed without any errors.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @NoelLincoln👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with model validations. ✔️ 
- Nice UI design. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/index.html.erb (hunk 1 above)
Comment (NoelLincoln) [reply app/views/categories/index.html.erb]: I just noticed, thank you. Fixed

**Review by Efiamotu-1 — APPROVED** [review]
Hi @NoelLincoln, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**